### DISCUSSION TITLE
.segment "DIRECTPAGE"

### DISCUSSION THREAD

**Discussion by jeffythedragonslayer** [cc65/cc65#1750] [Q&A]
Eyes & Litchy on page 49 write:

> a zero page which, renamed the **direct page**, can be relocated to any contiguous set of $100 bytes anywhere within the first 64K of memory (which in the case of the 65816 is anywhere in its address space).

Can there be a `.segment "DIRECTPAGE" ` for ca65?  That would make it easier to tell at a glance when code is meant to run on the 65816 or 65802.

**Comment by mrdudz** [comment]
I don't understand the question - you can name your segments however you want. And we dont have any predefined linker config for a 65816 target as far as i can see.

**Comment by pinobatch** [comment]
related: issues #194, #317, #498, and #881 (assembler-level support for direct page at $2000 or other non-$0000 addresses) and [mailing list discussion about adding `.DPAGE`](https://sourceforge.net/p/cc65/mailman/cc65-devel/thread/CAOHrKHZpoBuP4s0%2BoJE0N8CU0F5UOg0mfVm87dcVVw8ri%3DPRZw%40mail.gmail.com/#msg35176922)

**Comment by jeffythedragonslayer** [comment]
Yes I was also thinking `.DIRECTPAGE`

**Reply by polluks** [reply]
> Yes I was also thinking `.DIRECTPAGE`

Well, we already have `.P816`. What should your directive do? Some kind of `TCD`/`TDC`?

**Reply by jeffythedragonslayer** [reply]
I guess, what were you thinking about `TCD / TDC`?  Putting `.P816` above `.ZEROPAGE` isn't bad either.

**Comment by mrdudz** [comment]
Ok, as far as i understand this isnt about "segments" (as in linker config) at all - but some pseudoop that controls the behaviour of the assember.

As Oliver already said in one of the other issues, i am willing to merge a PR that implements the required changes. I wont work on this myself though (i barely know anything about 65816 to begin with)
